src/components/checkbox/checkbox-base.ts

```diff
@@ -42,6 +42,10 @@ export class IgcCheckboxBaseComponent extends EventEmitterMixin<
   @property({ type: Boolean, reflect: true })
   public invalid = false;
 
+  /** Controls the keyboard focus of the element. */
+  @property({ type: Boolean, reflect: true })
+  public focused = false;
```

Comment:
Can we make this an internal `@state` property and set the part attribute somewhere in the shadow DOM?
Can you achieve the styling with that approach?
Not really into exposing this as a public property which handles internal state...

Reply:
Done.